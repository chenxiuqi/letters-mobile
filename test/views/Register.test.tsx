import React from "react";
import { Button } from "@components";
import { RegisterScreen } from "@views";
import renderer from "react-test-renderer";
import fetchMock from "jest-fetch-mock";

const setup = (response = {}) => {
  const navigation = { navigate: jest.fn() };
  const element = renderer.create(<RegisterScreen navigation={navigation} />);
  const instance = element.getInstance();
  if (Object.keys(response).length > 0) {
    fetchMock.mockOnce(JSON.stringify(response));
  }
  return {
    element,
    instance,
  };
};

describe("Register screen", () => {
  it("should render", () => {
    const { element } = setup();
    const tree = element.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should have register button be disabled until all fields are valid", async () => {
    const { element, instance } = setup();
    const registerScreen = element.root;
    const registerButton = registerScreen.findAllByType(Button)[2];
    expect(registerButton.props.enabled).toBe(false);
    instance.devSkip();
    expect(registerButton.props.enabled).toBe(true);
  });
  it("should give an error and not submit on bad register", async () => {
    const { element, instance } = setup({
      type: "error",
      data: "email in use",
    });
    const registerScreen = element.root;
    const registerButton = registerScreen.findAllByType(Button)[2];
    instance.devSkip();
    await registerButton.props.onPress();
    //expect(instance.state.registered).toBe(false);
  });
  it("should successfully register a new user on good register", async () => {
    const { element, instance } = setup({
      data: {
        id: "6",
        firstName: "Mark",
        lastName: "Pekala",
        email: "mpekala@college.harvard.edu",
        cell: "6127038623",
        addresss1: "210 W Diamond Lake Road",
        country: "USA",
        zipcode: "55419",
        city: "Minneapolis",
        state: "MN",
      },
      type: "success",
    });
    const registerScreen = element.root;
    const registerButton = registerScreen.findAllByType(Button)[2];
    instance.devSkip();
    await registerButton.props.onPress();
    //expect(instance.state.registered).toBe(true);
  });
});
