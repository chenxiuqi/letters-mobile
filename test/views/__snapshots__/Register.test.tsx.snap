// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register screen should render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 100,
        "paddingTop": 50,
      }
    }
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 10,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#3490dc",
              "borderRadius": 50,
              "height": 100,
              "opacity": 1,
              "overflow": "hidden",
              "width": 100,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontStyle": "italic",
                "fontWeight": "400",
              },
              Object {
                "marginTop": 5,
              },
            ]
          }
        >
          Click to upload profile image.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D0E5FD",
            "borderColor": "#BEDAFC",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 15,
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontWeight": "600",
              },
              Object {
                "color": "#124181",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          We value your privacy. We will only use your address information as the return address for the mailed letter.
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginBottom": 10,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
              },
              Object {
                "color": "#3490dc",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Already have an account?
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3490dc",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
              Object {},
            ]
          }
        >
          Dev Skip
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="First Name"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Last Name"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 50,
                "marginVertical": 5,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <RCTScrollView
            keyboardShouldPersistTaps="never"
            scrollEnabled={false}
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <View>
              <TextInput
                allowFontScaling={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Cell Phone Number"
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "borderColor": "#282828",
                      "borderRadius": 4,
                      "borderWidth": 2,
                      "color": "#282828",
                      "flex": 1,
                      "fontSize": 18,
                      "height": 50,
                      "padding": 10,
                    },
                    Object {},
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Address Line 1"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Address Line 2"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Country"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Zip Code"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="City"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="State"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Email"
              rejectResponderTermination={true}
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 50,
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      >
        <RCTScrollView
          keyboardShouldPersistTaps="never"
          scrollEnabled={false}
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "borderColor": "#282828",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "color": "#282828",
                    "flex": 1,
                    "fontSize": 18,
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </RCTScrollView>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "grey",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
